pkg/bundle/resources.go

```diff
@@ -103,6 +103,15 @@ type Auth struct {
 }
 
 func chartURL(location *fleet.HelmOptions, auth Auth) (string, error) {
+
+	hasOCIURL, err := regexp.MatchString(`^oci:\/\/`, location.Chart)
+	if err != nil {
+		return "", err
+	}
+	if hasOCIURL {
+		return location.Chart, nil
+	}
+
```

Comment:
What do you try to achieve here? `Repo` should be empty in OCI urls as in my [example](https://github.com/rancher/fleet-examples/blob/master/single-cluster/helm-oci/fleet.yaml) so this method should return the location.Chart already without additional logic.

And if there is an issue in OCI chart handling, I would put it at least in a separate commit or better in a new pr - especially if it is not related.

Reply:
I don't think this is really an issue with the handling of OCI charts. The change in chartUrl to check for OCI urls is to prevent the code that follows to try to use the oci url as a chart name. Normally, chartUrl wouldn't be called because the function calling `chartUrl` has a check for `spec.Repo != ""` that prevents the call if it's an OCI chart. I think the only way a user would hit this currently would be if they supplied a Repo and an oci/go-getter chart.

The addition of `propagateHelmChartProperties` causes the calling function's check to be true in some cases. Instead of this change, I think there is a change that can be made to the calling check to make sure that chartUrl isn't called in that case.

`propagateHelmChartProperties` can also be updated to prevent those propagations from hitting this case, unless the user explicitly supplies a Repo and Chart. I think something similar to the current commented body of  `propagateHelmChartProperties`'s loop would solve this problem.

Reply:
You are right, it is better to prevent potential user issues by handling this automatically.